plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticator.java

```diff
@@ -201,6 +226,19 @@ private String lookupUserDistinguishedName(String user)
         return getOnlyElement(userDistinguishedNames);
     }
 
+    private void validatePassword(String userName, String password)
```

Comment:
Just like the above - I like this kind of method and I think that this kind of abstraction should be provided for each of the `ldapClient` usages, regardless of the fact whether this method is used in one or more places. Preferably all this kind of methods should be extracted to a separate class. 

This should set proper boundaries in the code for using this kind of third party's service client and allow others to extend it in an organised manner.